_hooks/pre-commit

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+#
+
+# A git hook script that checks for trailing whitespace
+# in your commits. Bypass it with the --no-verify option
+# to git-commit
+#
+# usage: make a soft link to this file,
+# e.g., ln -s ~/config/pre-commit.git.sh ~/some_project/.git/hooks/pre-commit
+
+if test "$(git diff --check --cached)"
```

Comment:
Oh, sorry I should have elaborated more. Basically, can we make the pre-commit hook identical to the Travis check, but maybe a little more abbreviated? So something like:

```
mdl README.md _posts _pages --style _linter/markdown-linter-style.rb && \
  bundle exec jekyll build && \
  bundle exec htmlproofer ./_site --only-4xx --check-html --file-ignore "/mlab_observatory/" \
    --disable_external
```

I'd like the pre-commit hook to complete in <15s, so that's why I'd like to remove the external check. But the goal is that users don't have to do a PR just to see if the build checks are passing and can get reasonably strong confidence just from the pre-commit hook.